### ISSUE TITLE
Inconsistent Metric Results in terramind_v1_base_sen1floods11.ipynb Testing

### ISSUE THREAD

**Issue by eterna621** [IBM/terramind#25]
Hi all! Thanks for the great work!

While executing terramind_v1_base_sen1floods11.ipynb in Google Colab, I observed inconsistencies in the resulting test metrics. This discrepancy appears when the fine-tuned model is tested in two different ways:

1. Passing the` datamodule` to the test function.

<img width="738" height="439" alt="Image" src="https://github.com/user-attachments/assets/0666b4e0-89a8-4042-a51c-e00ecdbdc8bf" />

2. Passing the `test_dataloader` directly.

<img width="1039" height="363" alt="Image" src="https://github.com/user-attachments/assets/78e3785c-86aa-407c-8717-1a05c50a3aad" />

I've included screenshots. This likely explains why the predictions after the third epoch look incorrect and don't match the TensorBoard results.

Could you please investigate this metric discrepancy?

**Comment by blumenstiel** [comment]
Hi @eterna621, thanks for reporting. This looks like an issue from Lightning/TerraTorch. @Joao-L-S-Almeida have you seen this before?

It's interesting that `test/loss` is missing in the second run. Often this is the case if you run with half precision and NaNs appear somewhere in the model (or input data). Can you report your second run with full precision?

**Comment by eterna621** [comment]
Thank you for your quick response. Here is the second run with full precision:

<img width="1039" height="364" alt="Image" src="https://github.com/user-attachments/assets/a103e152-3c91-4936-af31-7323151b2d00" />

**Comment by blumenstiel** [comment]
Thanks for testing! Agree, it must be something related to steps missing in the data loader that are applied in the datamodule. E.g., input normalization is handled not in transform but in another step in torchgeo/terratorch. This might also explain the accuracy close to 50%, which is basically random.

 @Joao-L-S-Almeida Ideally we find a solution for this issue, or if possible we can raise an error or warning if a dataloader is passed directly.
